### ISSUE TITLE
 1.24.0 fails to create js file with "LLVM ERROR: I->getOperand(0)->getType() == i64"

### ISSUE THREAD

**Issue by bobajeff** [emscripten-core/emscripten#2834]
I'm trying to convert a bitcode file, generated with Emscripten 1.24.0 (when compiling fastcomp), to a js file. I was able to successfully convert llvm-as.bc and llvm-dis.bc to js but clang.bc errors out with:

> LLVM ERROR: I->getOperand(0)->getType() == i64
> Traceback (most recent call last):
>   File "/home/bobajeff/Downloads/emscripten/emscripten-1.24.0/emscripten.py", line 1504, in <module>
>     _main(environ=os.environ)
>   File "/home/bobajeff/Downloads/emscripten/emscripten-1.24.0/emscripten.py", line 1492, in _main
>     temp_files.run_and_clean(lambda: main(
>   File "/home/bobajeff/Downloads/emscripten/emscripten-1.24.0/tools/tempfiles.py", line 39, in run_and_clean
>     return func()
>   File "/home/bobajeff/Downloads/emscripten/emscripten-1.24.0/emscripten.py", line 1500, in <lambda>
>     DEBUG_CACHE=DEBUG_CACHE,
>   File "/home/bobajeff/Downloads/emscripten/emscripten-1.24.0/emscripten.py", line 1387, in main
>     jcache=jcache, temp_files=temp_files, DEBUG=DEBUG, DEBUG_CACHE=DEBUG_CACHE)
>   File "/home/bobajeff/Downloads/emscripten/emscripten-1.24.0/emscripten.py", line 774, in emscript_fast
>     backend_output = open(temp_js).read()
> IOError: [Errno 2] No such file or directory: '/tmp/tmpsY3dSp.4.js'
> Traceback (most recent call last):
>   File "/home/bobajeff/Downloads/emscripten/emscripten-1.24.0/emcc", line 1257, in <module>
>     final = shared.Building.emscripten(final, append_ext=False, extra_args=extra_args)
>   File "/home/bobajeff/Downloads/emscripten/emscripten-1.24.0/tools/shared.py", line 1464, in emscripten
>     assert os.path.exists(filename + '.o.js'), 'Emscripten failed to generate .js'
> AssertionError: Emscripten failed to generate .js

I did a search and found [a very similar error occurred with 1.16.0.](https://github.com/juj/emsdk/issues/5). However, my version of Emscripten is a manual build instead of the SDK.

<h2>Duplicating the error</h2>

Because the build process is complicated and long, here's a link to the clang.bc file:
http://www.mediafire.com/download/bnqend7557e97nj/clang.bc

Just run the following where you put the clang.bc file.

> emcc clang.bc -o clang.js

<h2>Building from scratch </h2> 

The commands used to generated the file were adapted from the [llvm.js' Readme.md](https://github.com/kripken/llvm.js/blob/master/README.md). After downloading and extracting [emscripten-fastcomp-1.24.0](https://codeload.github.com/kripken/emscripten-fastcomp/tar.gz/1.24.0) and [emscripten-fastcomp-clang-1.24.0](https://codeload.github.com/kripken/emscripten-fastcomp-clang/tar.gz/1.24.0) and [my patch](https://gist.github.com/bobajeff/4e6b74a7316c647a4cec). 

> mv emscripten-fastcomp-clang-1.24.0 emscripten-fastcomp-1.24.0/tools/clang
> cd emscripten-fastcomp-clang-1.24.0/
> patch -p1 < ../emscripten_build.diff
> mkdir build
> cd build/
> emconfigure ../configure --enable-optimized --disable-jit --disable-threads --disable-pthreads --enable-targets=x86,js
> emmake make

You'll get permission denied errors when it tries to run the bitcode files. As in:

> make[1]: execvp: /home/bobajeff/Downloads/fastcomp.js/emscripten-fastcomp-1.24.0/build/Release+Asserts/bin/llvm-tblgen: Permission denied
> /bin/sh: 1: /home/bobajeff/Downloads/fastcomp.js/emscripten-fastcomp-1.24.0/build/Release+Asserts/bin/llvm-config: Permission denied
> make[5]: execvp: /home/bobajeff/Downloads/fastcomp.js/emscripten-fastcomp-1.24.0/build/Release+Asserts/bin/clang-tblgen: Permission denied

When they come up replace the respective bitcode file with it's native equivalent and edit the relevant Makefiles so they don't get overwritten and re-run emmake make after each one respectively. _Alternatively, instead of manually editing the Makefiles apply [these patches](https://gist.github.com/bobajeff/c1529513ef9bf5728a3a) for their respective file._

So basically:

> patch -p1 < ../llvm-tblgen-fix.diff
> cp /path/to/native/llvm-tblgen emscripten-fastcomp-1.24.0/build/Release+Asserts/bin/
> emmake make
> 
> patch -p1 < ../llvm-config-fix.diff
> cp /path/to/native/llvm-config emscripten-fastcomp-1.24.0/build/Release+Asserts/bin/
> emmake make
> 
> patch -p1 < ../clang-tblgen-fix.diff
> cp /path/to/native/clang-tblgen emscripten-fastcomp-1.24.0/build/Release+Asserts/bin/
> emmake make

Once everything is built rename the clang bitcode file in emscripten-fastcomp-1.24.0/build/Release+Asserts/bin to clang.bc. Then in that directory run:

> emcc clang.bc -o clang.js

**Comment by bobajeff** [comment]
Update:
I compiled with the current master branch of emscripten and fastcomp (llvm4.3) and clang.bc coverts to js without assertion errors (I didn't fiddle with the object files this time). So it appears to have been fixed. I'm not sure though as any number of things could've changed in the version of clang I compiled. I will test a little more before I close this.
